scene/gui/link_button.cpp

```diff
@@ -29,24 +29,49 @@
 /*************************************************************************/
 
 #include "link_button.h"
-
+#include "core/os/os.h"
 #include "core/translation.h"
 
 void LinkButton::set_text(const String &p_text) {
 	if (text == p_text) {
 		return;
 	}
+
 	text = p_text;
-	xl_text = tr(p_text);
-	update();
 	_change_notify("text");
+	update_text();
+}
+
+void LinkButton::set_url(const String &p_url) {
+	if (url == p_url) {
+		return;
+	}
+
+	url = p_url;
+	_change_notify("url");
+	update_text();
+}
+
+void LinkButton::update_text() {
+	xl_text = tr((text.is_empty()) ? url : text);
```

Comment:
```suggestion
	xl_text = text.is_empty() ? url : tr(text);
```

Probably should be like this, there's no reason to translate URL.

Reply:
That's a good point, will do. I also noticed that the `is_empty` method doesn't exist for Strings in G3 (but I saw it in the G4 related issue?), so I really need to make a few corrections anyway

Reply:
In 3.x it's called `empty()`.

Reply:
@bruvzg Actually, tr(...) doesn't return a String but a StringName, so the ternary operator doesn't work:
```
if (text.empty()) {
	xl_text = url;
}
else {
	xl_text = tr(text);
}
```